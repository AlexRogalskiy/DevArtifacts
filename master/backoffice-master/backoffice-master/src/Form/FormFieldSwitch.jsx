import React from 'react'
import PropTypes from 'prop-types'

import {
  FormControlLabel,
  FormHelperText,
  Switch,
} from '@material-ui/core'

class FormFieldSwitch extends React.Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    handleChange: PropTypes.func.isRequired,
    helperText: PropTypes.string,
    classNames: PropTypes.arrayOf(PropTypes.string),
    isDisabled: PropTypes.bool,
  }

  static defaultProps = {
    helperText: null,
    classNames: [],
    isDisabled: false,
  }

  state = {
    value: false,
  }

  handleChange(id) {
    const func = this.props.handleChange(id)

    return () => {
      const newValue = !this.state.value

      this.setState({
        value: newValue,
      })

      return func({
        target: {
          value: newValue,
        },
      })
    }
  }

  render() {
    const {
      id,
      title,
      helperText,
      isDisabled,
      classNames,
    } = this.props

    return (
      <div className={classNames.join(' ')}>
        <FormControlLabel
          disabled={isDisabled}
          control={
            <Switch
              checked={this.state.value}
              onChange={this.handleChange(id)}
              value={id}
              color="primary"
            />
          }
          label={title}
        />

        {helperText ? (
          <FormHelperText
            disabled={isDisabled}
          >
            {helperText}
          </FormHelperText>
        ) : null}
      </div>
    )
  }
}

export default FormFieldSwitch
