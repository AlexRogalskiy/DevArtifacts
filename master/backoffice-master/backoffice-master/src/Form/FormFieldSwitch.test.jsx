import React from 'react'
import { mount } from 'enzyme'

import FormFieldSwitch from './FormFieldSwitch'

it('renders correctly', () => {
  const tree = mount((
    <FormFieldSwitch
      id="switch"
      title="Title"
      handleChange={() => () => {}}
    />
  ))

  expect(tree).toMatchSnapshot()
})

it('renders disabled', () => {
  const tree = mount((
    <FormFieldSwitch
      id="switch"
      title="Title"
      handleChange={() => () => { }}
      isDisabled
    />
  ))

  expect(tree).toMatchSnapshot()
})

it('renders with helper text', () => {
  const tree = mount((
    <FormFieldSwitch
      id="switch"
      title="Title"
      handleChange={() => () => { }}
      helperText="test"
    />
  ))

  expect(tree).toMatchSnapshot()
})
