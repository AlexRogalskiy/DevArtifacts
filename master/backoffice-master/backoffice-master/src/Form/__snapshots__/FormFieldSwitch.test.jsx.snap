// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className=""
>
  <label
    className="MuiFormControlLabel-root-1"
  >
    <span
      className="MuiSwitch-root-5"
    >
      <span
        className="MuiButtonBase-root-24 MuiIconButton-root-18 MuiSwitchBase-root-14 MuiSwitch-switchBase-8 MuiSwitch-colorPrimary-10"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label-23"
        >
          <span
            className="MuiSwitch-icon-6"
          />
          <input
            checked={false}
            className="MuiSwitchBase-input-17"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
            value="switch"
          />
        </span>
        <span
          className="MuiTouchRipple-root-27"
        />
      </span>
      <span
        className="MuiSwitch-bar-13"
      />
    </span>
    <span
      className="MuiTypography-root-34 MuiTypography-body1-43 MuiFormControlLabel-label-4"
    >
      Title
    </span>
  </label>
</div>
`;

exports[`renders disabled 1`] = `
<div
  className=""
>
  <label
    className="MuiFormControlLabel-root-1 MuiFormControlLabel-disabled-3"
  >
    <span
      className="MuiSwitch-root-5"
    >
      <span
        className="MuiButtonBase-root-24 MuiButtonBase-disabled-25 MuiIconButton-root-18 MuiIconButton-disabled-22 MuiSwitchBase-root-14 MuiSwitch-switchBase-8 MuiSwitch-colorPrimary-10 MuiSwitchBase-disabled-16 MuiSwitch-disabled-12"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
      >
        <span
          className="MuiIconButton-label-23"
        >
          <span
            className="MuiSwitch-icon-6"
          />
          <input
            checked={false}
            className="MuiSwitchBase-input-17"
            disabled={true}
            onChange={[Function]}
            type="checkbox"
            value="switch"
          />
        </span>
      </span>
      <span
        className="MuiSwitch-bar-13"
      />
    </span>
    <span
      className="MuiTypography-root-34 MuiTypography-body1-43 MuiFormControlLabel-label-4 MuiFormControlLabel-disabled-3"
    >
      Title
    </span>
  </label>
</div>
`;

exports[`renders with helper text 1`] = `
<div
  className=""
>
  <label
    className="MuiFormControlLabel-root-1"
  >
    <span
      className="MuiSwitch-root-5"
    >
      <span
        className="MuiButtonBase-root-24 MuiIconButton-root-18 MuiSwitchBase-root-14 MuiSwitch-switchBase-8 MuiSwitch-colorPrimary-10"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label-23"
        >
          <span
            className="MuiSwitch-icon-6"
          />
          <input
            checked={false}
            className="MuiSwitchBase-input-17"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
            value="switch"
          />
        </span>
        <span
          className="MuiTouchRipple-root-27"
        />
      </span>
      <span
        className="MuiSwitch-bar-13"
      />
    </span>
    <span
      className="MuiTypography-root-34 MuiTypography-body1-43 MuiFormControlLabel-label-4"
    >
      Title
    </span>
  </label>
  <p
    className="MuiFormHelperText-root-59"
  >
    test
  </p>
</div>
`;
